docs/shader_object_layer.md

```diff
@@ -0,0 +1,138 @@
+<!-- markdownlint-disable MD041 -->
+<p align="left"><img src="https://vulkan.lunarg.com/img/NewLunarGLogoBlack.png" alt="LunarG" width=263 height=113 /></p>
+
+Copyright &copy; 2015-2023 LunarG, Inc.
+
+[![Creative Commons][3]][4]
+
+[3]: https://i.creativecommons.org/l/by-nd/4.0/88x31.png "Creative Commons License"
+[4]: https://creativecommons.org/licenses/by-nd/4.0/
+
+
+# VK\_LAYER\_KHRONOS\_shader_object
+The `Shader object` extension Layer implements the `VK_EXT_shader_object` extension.
+By default, it will disable itself if the underlying driver provides the extension.
+
+The `VK_EXT_shader_object` extension requires the `VK_KHR_maintenance2` and `VK_KHR_dynamic_rendering` extensions, this layer will not work on devices that do not implement them.
+
+## Configuring the Shader object layer
+
+For an overview of how to configure layers, refer to the [Layers Overview and Configuration](https://vulkan.lunarg.com/doc/sdk/latest/windows/layer_configuration.html) document.
+
+The Shader object Layer settings are documented in the [Layer Details](https://vulkan.lunarg.com/doc/sdk/latest/windows/shader_object_layer.html#user-content-layer-details) section below.
+
+The Shader object Layer can also be enabled and configured using vkconfig. See the [vkconfig](https://vulkan.lunarg.com/doc/sdk/latest/windows/vkconfig.html) documentation for more information.
```

Comment:
Should we use VK_LAYER_Khronos_shader_object instead of "The Shader object layer"?

Reply:
This was pretty much copied from sync2 documentation, same wording is used there. But I can change it if this is preferred

Reply:
It's not important or critical. Just a nit